core/src/main/clojure/xtdb/pgwire.clj

```diff
@@ -1505,32 +1507,38 @@
 
 (defn parse
   "Responds to a msg-parse message that creates a prepared-statement."
-  [{:keys [conn-state]} {:keys [query]}]
+  [{:keys [conn-state ^Counter query-error-counter]} {:keys [query]}]
 
   (let [{:keys [session watermark-tx-id]} @conn-state
         {:keys [^Clock clock], session-parameters :parameters} session]
 
-    (interpret-sql query {:default-tz (.getZone clock)
-                          :watermark-tx-id watermark-tx-id
-                          :session-parameters session-parameters})))
-
-(defn- prep-stmt [{:keys [node, conn-state] :as conn} {:keys [statement-type] :as stmt} {:keys [param-oids]}]
-  (let [{:keys [session watermark-tx-id]} @conn-state
-        {:keys [^Clock clock]} session
-        fallback-output-format (get-in session [:parameters "fallback_output_format"])
-        param-types (map types/pg-types-by-oid param-oids)]
-
-    (if (or (contains? #{:query :execute} statement-type)
-            (and (= :prepare statement-type)
-                 (= :query (:inner-statement-type stmt))))
-      (let [param-col-types (mapv :col-type param-types)]
-        (when (some nil? param-col-types)
-          (throw (ex-info "unsupported param-types in query"
-                          {::client-error (err-protocol-violation (str "Unsupported param-types in query: "
-                                                                       (pr-str (->> param-types
-                                                                                    (into [] (comp (filter (comp nil? :col-type))
-                                                                                                   (map :typname)
-                                                                                                   (distinct)))))))})))
+    (try
+      (interpret-sql query {:default-tz (.getZone clock)
+                            :watermark-tx-id watermark-tx-id
+                            :session-parameters session-parameters})
+      (catch Exception e
+        (when query-error-counter
+          (.increment query-error-counter))
```

Comment:
We keep this error counter because an error here does never go through the node.

Reply:
Let's scrap this one for now until we come back to adding transaction-error counters. So now error counting at all at this place.